### PR TITLE
adding stop signals

### PR DESCRIPTION
**Pull request by SethHollandsworth**
adding field from OCI spec that lets you customize which signal is used as "stop"

### CODE REVIEW

File: src/confcom/azext_confcom/security_policy.py [hunk 1]
```diff
@@ -350,6 +350,7 @@ def _policy_serialization(self, pretty_print=False) -> str:
             return pretty_print_func(policy)
         return print_func(policy)
 
+    # pylint: disable=R0914, R0915
```
Comment (hgarvison) [inline src/confcom/azext_confcom/security_policy.py:353]: Is this supposed to be here?

File: src/confcom/azext_confcom/tests/latest/test_confcom_arm.py [hunk 2]
```diff
@@ -5293,4 +5293,142 @@ def test_arm_template_security_context_seccomp_profile_missing_syscalls(self):
...
+
+    def test_stop_signal(self):
+        regular_image_json = json.loads(
+            self.aci_arm_policy.get_serialized_output(
+                output_type=OutputType.RAW, rego_boilerplate=False
+            )
+        )
+        # check for the signal for SIGQUIT. this is part of the nginx image
+        self.assertTrue(
+            3 in
```
Comment (hgarvison) [inline src/confcom/azext_confcom/tests/latest/test_confcom_arm.py:5432]: I would also check that it gets converted to a string properly by checking that SIGQUIT is in the policy

File: src/confcom/azext_confcom/security_policy.py (hunk 1 above)
Comment (SethHollandsworth) [reply src/confcom/azext_confcom/security_policy.py:353]: yeah it violates too many statements and too many local variables but refactoring, in my opinion, would look worse

File: src/confcom/azext_confcom/tests/latest/test_confcom_arm.py (hunk 2 above)
Comment (SethHollandsworth) [reply src/confcom/azext_confcom/tests/latest/test_confcom_arm.py:5432]: I think it's only supposed to be a number in the policy

Comment (hgarvison) [reply src/confcom/azext_confcom/tests/latest/test_confcom_arm.py:5432]: I thought Bryce had said the OCI policy spec only considered the strings during standup

Comment (SethHollandsworth) [reply src/confcom/azext_confcom/tests/latest/test_confcom_arm.py:5432]: [looks like you're right](https://github.com/opencontainers/image-spec/blob/b74f37875da0de3a68497790c0c74a996c8a6211/schema/config_test.go#L185C33-L185C33). I'll test to see if Docker does that translation for you or if it'll put a number in the image manifest

Comment (SethHollandsworth) [reply src/confcom/azext_confcom/tests/latest/test_confcom_arm.py:5432]: talked with Matthew and he said the policy is expecting a number for the signal, not a name. I think we should be good as-is

**Review by hgarvison — APPROVED** [review]

**Review by hgarvison — APPROVED** [review]

**Pull request merged**